voronoi_game.py

```diff
@@ -103,29 +107,15 @@ def __get_player_move(self):
     data = client_response.split()
     return int(data[0]), int(data[1])
 
+  def compute_pull(self, row, col):
+    return np.reciprocal(np.sqrt(np.square(self.row_numbers - row) + np.square(self.col_numbers - col) + 0.1e-30))
+
   def __update_scores(self, move_row, move_col):
-    # note: score ignores stones, because each player has the same number of stones
-    for row in range(self.grid_size):
-      for col in range(self.grid_size):
-        # avoid division by 0
-        if (row == move_row and col == move_col):
-          continue
-        # update current player's pull
-        d = self.__compute_distance(row, col, move_row, move_col)
-        self.pull[self.current_player][row][col] += float(float(1) / (d*d))
-
-        old_occupier = self.score_grid[row][col]
-        # if the cell has not been claimed by any player
-        if old_occupier == 0:
-          self.score_grid[row][col] = self.current_player + 1
-          self.scores[self.current_player] += 1
-        # if the cell is claimed by some other player
-        elif old_occupier - 1 != self.current_player:
-          # and current player now has a greater pull
-          if self.pull[self.current_player][row][col] > self.pull[old_occupier - 1][row][col]:
-            self.score_grid[row][col] = self.current_player + 1
-            self.scores[old_occupier - 1] -= 1
-            self.scores[self.current_player] += 1
+    self.pull[self.current_player] = self.pull[self.current_player] + self.compute_pull(move_row, move_col)
```

Comment:
You're adding the same pull (which also seems to be weirdly/incorrectly calculated above) to all the points in the grid, but the pull changes for every point that's not on the same "circle" so that just wouldn't work.

Reply:
It's not adding the same pull to every point in the grid. That is where row_numbers and col_numbers come in picture.